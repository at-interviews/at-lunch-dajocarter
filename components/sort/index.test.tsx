import { fireEvent, render, screen } from "@testing-library/react";
import SortButton from ".";

const handleSort = jest.fn()
describe('SortButton component', () => {
  it('matches snapshot before clicking Sort button', () => {
    const { container } = render(<SortButton handleSort={handleSort} />)

    expect(container).toMatchSnapshot()
  })
  it('matches snapshot after clicking Sort button', () => {
    const { container } = render(<SortButton handleSort={handleSort} />)
    fireEvent.click(screen.getByText('Sort'))

    expect(container).toMatchSnapshot()
  })
  it('matches snapshot after applying Sort', () => {
    const { container } = render(<SortButton handleSort={handleSort} />)
    fireEvent.click(screen.getByText('Sort'))
    fireEvent.click(screen.getByLabelText('Ratings: Low to High'))
    fireEvent.click(screen.getByText('Apply'))

    expect(container).toMatchSnapshot()
  })
  it('should handle form submission', () => {
    render(<SortButton handleSort={handleSort} />)
    fireEvent.click(screen.getByText('Sort'))
    fireEvent.click(screen.getByLabelText('Ratings: Low to High'))
    fireEvent.click(screen.getByText('Apply'))

    expect(handleSort).toHaveBeenCalledWith('ASC')
  })
})