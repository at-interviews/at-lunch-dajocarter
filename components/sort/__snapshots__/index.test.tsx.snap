// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SortButton component matches snapshot after applying Sort 1`] = `
<div>
  <div
    class="sortButtonContainer"
  >
    <button
      class="sortButton inactiveButton"
    >
      Sort
    </button>
    <div
      class="tooltipContainer hiddenTooltip"
    >
      <form
        class="sortForm"
      >
        <input
          id="high-low"
          name="sort-by-ratings"
          type="radio"
          value="DESC"
        />
        <label
          for="high-low"
        >
          Ratings: High to Low
        </label>
        <input
          id="low-high"
          name="sort-by-ratings"
          type="radio"
          value="ASC"
        />
        <label
          for="low-high"
        >
          Ratings: Low to High
        </label>
        <button
          class="applySort"
          type="submit"
        >
          Apply
        </button>
      </form>
    </div>
  </div>
  <div
    class="overlay hiddenOverlay"
  />
</div>
`;

exports[`SortButton component matches snapshot after clicking Sort button 1`] = `
<div>
  <div
    class="sortButtonContainer"
  >
    <button
      class="sortButton activeButton"
    >
      Sort
    </button>
    <div
      class="tooltipContainer visibleTooltip"
    >
      <form
        class="sortForm"
      >
        <input
          id="high-low"
          name="sort-by-ratings"
          type="radio"
          value="DESC"
        />
        <label
          for="high-low"
        >
          Ratings: High to Low
        </label>
        <input
          id="low-high"
          name="sort-by-ratings"
          type="radio"
          value="ASC"
        />
        <label
          for="low-high"
        >
          Ratings: Low to High
        </label>
        <button
          class="applySort"
          type="submit"
        >
          Apply
        </button>
      </form>
    </div>
  </div>
  <div
    class="overlay activeOverlay"
  />
</div>
`;

exports[`SortButton component matches snapshot before clicking Sort button 1`] = `
<div>
  <div
    class="sortButtonContainer"
  >
    <button
      class="sortButton inactiveButton"
    >
      Sort
    </button>
    <div
      class="tooltipContainer hiddenTooltip"
    >
      <form
        class="sortForm"
      >
        <input
          id="high-low"
          name="sort-by-ratings"
          type="radio"
          value="DESC"
        />
        <label
          for="high-low"
        >
          Ratings: High to Low
        </label>
        <input
          id="low-high"
          name="sort-by-ratings"
          type="radio"
          value="ASC"
        />
        <label
          for="low-high"
        >
          Ratings: Low to High
        </label>
        <button
          class="applySort"
          type="submit"
        >
          Apply
        </button>
      </form>
    </div>
  </div>
  <div
    class="overlay hiddenOverlay"
  />
</div>
`;
